### PR TITLE
Enhancements to deployment process

### PR DESCRIPTION
**Pull request by leftclickben**
Actually covers:
- RWAHS-81 Create shortcut for deployment script that builds and deploys in one step
- RWAHS-98 Allow deployment script to use the latest release if none specified
- RWAHS-75 Deployment needs to symlink in setup.php and the media directory
- RWAHS-97 Gracefully handle deployment of release already deployed

### CODE REVIEW

File: include/rwahs/tasks/build [hunk 1]
```diff
@@ -103,12 +101,6 @@ function build_runTask {
...
     echo "Running 'build' task for component '${component}'..."
-    apiJson=$(printf "${apiJsonTemplate}" "${releaseName}" "${targetCommitish}" "${releaseName}" "${releaseDate}")
-    curl \
-        --request POST \
-        --user-agent "RWAHS Deployment Script" \
-        --header "Accept: application/vnd.github.v3+json" \
-        --header "Authorization: token ${RWAHS_ACCESS_TOKEN}" \
-        --data "${apiJson}" \
-        "${ghApiRoot}/repos/${repo}/releases"
+    github_createRelease "${RWAHS_ACCESS_TOKEN}" "${repo}" "${releaseName}" "${targetCommitish}" "Release tagged at ${releaseDate}" >/dev/null 2>&1
```
Comment (kehh) [inline include/rwahs/tasks/build]: are you sure you want to redirect `STDERR`?

File: include/util/github [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+#
+# util/github - Function definitions for interacting with the GitHub REST API.
+#
+
+set -e
+set -u
+
+# Output the root URL of the GitHub API.
+function github_rootUrl {
```
Comment (kehh) [inline include/util/github:10]: so global functions to avoid global variables?

**Comment by kehh** [comment]
two minor comments

File: include/util/github (hunk 2 above)
Comment (leftclickben) [inline include/util/github:10]: Yes.

**Comment by leftclickben** [comment]
Fixed redirection.

**Pull request merged**